pkg/util/framer/framer.go

```diff
@@ -131,13 +131,13 @@ func NewJSONFramedReader(r io.ReadCloser) io.ReadCloser {
 func (r *jsonFrameReader) Read(data []byte) (int, error) {
 	// Return whatever remaining data exists from an in progress frame
 	if n := len(r.remaining); n > 0 {
-		if n <= cap(data) {
+		if n <= len(data) {
```

Comment:
The contract of io.Reader/Writer is that callers deal with data as a
buffer, with the length being the size of the buffer. Using cap was
just wrong because it means the caller won't understand it (since read
can't change the slice extents, just the contents).